util/common/traces.go

```diff
@@ -0,0 +1,98 @@
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/util/awsservice"
+	"github.com/aws/aws-sdk-go-v2/service/xray/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type TraceConfig struct {
+	Interval    time.Duration
+	Annotations map[string]interface{}
+	Metadata    map[string]map[string]interface{}
+	Attributes  []attribute.KeyValue
+}
+type TracesTestInterface interface {
+	StartSendingTraces(ctx context.Context) error
+	StopSendingTraces()
+	Generate(ctx context.Context) error
+	validateSegments(t *testing.T, segments []types.Segment, config *TraceConfig)
+	GetTestCount() (int, int)
+	GetAgentConfigPath() string
+	GetGeneratorConfig() *TraceConfig
+	GetContext() context.Context
+	GetAgentRuntime() time.Duration
+	GetName() string
+}
+
+func TraceTest(t *testing.T, traceTest TracesTestInterface) error {
```

Comment:
Should call
```go
t.Helper()
```
at the start of this helper function so it'll map back to the calling function when printing out the failures.